packages/dd-trace/src/appsec/user_tracking.js

```diff
@@ -162,7 +159,36 @@ function trackLogin (framework, login, user, success, rootSpan) {
   return waf.run({ persistent })
 }
 
+function trackUser (user, rootSpan) {
+  if (!collectionMode || collectionMode === 'disabled') return
+
+  const userId = getUserId(user)
+  if (!userId) {
+    log.error('[ASM] No valid user ID found in AppSec trackUser')
+    telemetry.incrementMissingUserIdMetric('passport', 'authenticated_request')
+    return
+  }
+
+  rootSpan.setTag('_dd.appsec.usr.id', userId)
+
+  const isSdkCalled = rootSpan.context()._tags['_dd.appsec.user.collection_mode'] === 'sdk'
+  // do not override SDK
+  if (!isSdkCalled) {
+    rootSpan.addTags({
+      'usr.id': userId,
```

Comment:
sometimes we use `'usr.id'` and others `[addresses.USER_ID]`

Reply:
one is a span tag and the other is an address but...

Reply:
yes, one is a tag, and the other is a waf address, they should not be mixed. Although I agree the tags could use an enum instead of being hardcoded. See my comment above